### PR TITLE
blocks editing of renewal drafts at the controller level with an alert message

### PR DESCRIPTION
**Pull request by saikumar9**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [x] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [ ] There are no inline styles added
- [ ] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [x] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: [Jolly Podgers - 188067357](https://www.pivotaltracker.com/story/show/188067357)

# A brief description of the changes

Current behavior: The renewal draft applications are currently editable as this is **NOT blocked** at the Financial Assistance component endpoints level.

New behavior: The renewal draft applications will **NOT** be editable as this is **now blocked** at the Financial Assistance component endpoints level.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

File: components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb [hunk 1]
```diff
@@ -6,21 +6,52 @@ class ApplicationsController < FinancialAssistance::ApplicationController
 
     before_action :set_current_person
     before_action :set_family
-    before_action :find_application, :except => [:index, :index_with_filter, :new, :review, :raw_application]
+    before_action :find_application, :except => [:index, :index_with_filter, :new]
```
Comment (vkghub) [inline components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb:9]: The :review and :raw_application action methods already include queries to find the application. Removing these from the except list will result in querying the application twice, which can increase page load time.

I recommend either removing the application query from the respective methods or raising a ticket to address this refactor.

**Review by vkghub — CHANGES_REQUESTED** [review]

File: components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb (hunk 1 above)
Comment (saikumar9) [reply components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb:9]: Good catch. I will remove the duplicate queries.

**Review by vkghub — DISMISSED** [review]
LGTM

**Review by saipraveen18 — APPROVED** [review]
LGTM 👍

**Review by vkghub — APPROVED** [review]

**Pull request merged**